pkg/gpu/probe.go

```diff
@@ -192,30 +206,14 @@ func (p *Probe) Close() {
 func (p *Probe) GetAndFlush() (*model.GPUStats, error) {
 	now, err := ddebpf.NowNanoseconds()
 	if err != nil {
-		return nil, fmt.Errorf("getting current time: %w", err)
+		return nil, fmt.Errorf("error getting current time: %w", err)
 	}
 
-	stats := model.GPUStats{}
-	for key, handler := range p.consumer.streamHandlers {
-		currData := handler.getCurrentData(uint64(now))
-		pastData := handler.getPastData(true)
-
-		if currData != nil {
-			currData.Key = key
-			stats.CurrentData = append(stats.CurrentData, currData)
-		}
-
-		if pastData != nil {
-			pastData.Key = key
-			stats.PastData = append(stats.PastData, pastData)
-		}
-
-		if handler.processEnded {
-			delete(p.consumer.streamHandlers, key)
-		}
-	}
+	stats := p.generator.getStats(now)
+	p.generator.cleanupFinishedAggregators()
+	p.consumer.cleanupHandlersMarkedFinished()
```

Comment:
In this case I'd rather leave this without defer, as we don't want to clean up anything in case of error. Right now it doesn't matter as there are no possible errors, but I feel leaving it like this makes it less likely we introduce that problem in the future.

Reply:
that's fine, if defer doesn't suit the logic flow. but let's still wrap those with `probe.clean()` to handle all cleanup in one place